src/__tests__/global-container.test.ts

```diff
@@ -334,6 +334,16 @@ test("passes through the given params", () => {
   expect(instance.c).toBe(c);
 });
 
+test("works with abstract class", () => {
+  @injectable()
+  abstract class MyAbsClass {
+    protected constructor() { }
+  }
+
+  const instance = globalContainer.resolve(MyAbsClass);
```

Comment:
I don't think this should work - this would allow you to create an instance of an abstract class, which I don't think is desirable.

Reply:
I removed the injectable test as the autoInjectable test provides a more accurate case